src/asm/symbol.c

```diff
@@ -128,6 +128,22 @@ static char const *Callback__FILE__(void)
 	return buf;
 }
 
+static char const *Callback__SCOPE__(void)
+{
+	static char buf[MAXSYMLEN + 1];
+
+	if (!labelScope)
+		fatalerror("__SCOPE__ does not make sense in main scope\n");
+
+	size_t n = strlen(labelScope);
+
+	if (n > sizeof(buf) - 1)
```

Comment:
Can we assume `labelScope` will be a valid label, and omit this check?

Reply:
It should be; I would still make this an `assert` because debug sanity checks don't hurt.